threat_intel/util/http.py

```diff
@@ -273,10 +274,31 @@ def _wait_for_response(self, requests, to_json):
             A list of dicts if to_json, a list of grequest.response otherwise
         """
         all_responses = []
-
-        for request, response in zip(requests, grequests.map(requests)):
+        
+        retry = 0
+        max_retry = 10
+        responses = None
+        while retry <= max_retry and responses is None:
+            try:
+               retry = retry + 1
+               responses = grequests.map(requests)
+                # test our responses
+                for response in responses:
+                   # for some odd conditions, requests will neither return an object nor raise an error
+                   if not response:
+                       #raise an error so we trigger a retry
+                       responses = None
+                       raise ConnectionError('Batch of requests had an empty response')
+            except:
+                pass
+        
+        if responses is None:
+            raise ConnectionError('Unable to complete batch of requests within max_retry retries')
+
+        for request, response in zip(requests, responses):
             if not response:
```

Comment:
I don't think this would work. See if you are zipping request and response, you only get until the values they both match, for example: 
>>> a= [1,2,3,4]
>>> b=['a', 'b', 'c']
>>> for request, response in zip(a,b):
...       print request, response
1 a
2 b
3 c
>>>
So if there is more values in requests than in response, you will not get in this zip. This means that if not response will never happens. Maybe you don't want to zip them, but instead just loop over their values.